jormungandr/src/blockchain/chain.rs

```diff
@@ -290,12 +287,15 @@ impl Blockchain {
         &self.storage
     }
 
-    pub fn branches(&self) -> &Branches {
-        &self.branches
-    }
-
-    pub fn branches_mut(&mut self) -> &mut Branches {
-        &mut self.branches
+    pub async fn branches(&self) -> Result<Vec<Branch>> {
+        futures::stream::iter(self.storage().get_branches()?)
+            // FIXME: this should always return a valid ref, as the branches
```

Comment:
can this happen? wouldn't a `Vec<Result<_, _>>` make more sense? Although for the fund7 branch I guess it doesn't really matter.

Reply:
> wouldn't a Vec<Result<_, _>> make more sense?

Partly yes, but I think it will be just more cumbersome to use. Also because I expect errors not to depend on the specific branch but rather on external factors (e.g. spurious storage failures)